### DISCUSSION TITLE
std::cout on MOOSE

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#26785] [Q&A General] [answered]
All,
I am using a PenaltyInterfaceDiffusion type of an object at an interface. I am trying to see the fluctuations of value using “std::cout << “ time = “ << _t << “ u_value = “ << u_value << std::endl;”
where “u_value “ is the variable name of interest here.

Then, I get like:
time = 100 u_value = (-3.62547e-06,{})
.
.
 time = 100 u_value = (0.454298,{(7494,3.90874e-11), (7497,1160.03), (7704,1.45876e-10), (7707,4329.28), (16830,0), (16831,-2.50755e-05), (16836,0), (16837,-9.35832e-05)})
.
.

“time = 100 u_value = (-3.62547e-06,{})” seems that u_value = -3.62547e-06 at time = 100.
What about
 time = 100 u_value = (0.454298,{(7494,3.90874e-11), (7497,1160.03), (7704,1.45876e-10), (7707,4329.28), (16830,0), (16831,-2.50755e-05), (16836,0), (16837,-9.35832e-05)})?

Are 7494, 7497, 7704, 7707, 16830, 16831, 16836, 16837 the quadrature point index?
My model is 2D and why 8 points?
What are the values to the right of these?
Please help.
S. Thomas

**Comment by GiudGiud** [comment]
Hello

No these are the automatic differentiation indices. They are actually the derivatives of the value with regards to rhe degrees of freedom, and the indices (the big integers because they are global indices) of these dofs

Guillaume

**Reply by styyokuda** [reply]
Thank you very much, Guillaume , as always.
So, u_value = -3.62547e-06 for the first one, and u_value = 0.454298  for the second one are the values I have, aren't they?
Please confirm.
S. Thomas

**Reply by GiudGiud** [reply, marked as answer]
exactly
